docs/documentation/siddhi-architecture.md

```diff
@@ -94,237 +94,235 @@ Following diagram depicts the execution flow within a Siddhi App Runtime.
+and [AttributeAggregators](https://github.com/wso2/siddhi/blob/master/modules/siddhi-core/src/main/java/org/wso2/siddhi/core/query/selector/attribute/aggregator/AttributeAggregator.java)
 
-Current Events: Events that are newly arriving to a query from streams
+To achieve temporal processing,Siddhi uses the following four type of events: 
 
-Expired Events: Events that have expired from a window. 
+* **Current Events**: Events that are newly arriving to a query from streams.
 
-Timer Events: Events informing query about an update of execution time. Usually these events are generated by Schedulers. 
+* **Expired Events**: Events that have expired from a window. 
+
+* **Timer Events**: Events that inform the query about an update of an execution time. These events are usually generated by schedulers. 
  
-Reset Events: Events that resets Siddhi query states.
+* **Reset Events**: Events that resets the Siddhi query states.
  
 In Siddhi, when an event comes into a [WindowProcessor](https://github.com/wso2/siddhi/blob/master/modules/siddhi-core/src/main/java/org/wso2/siddhi/core/query/processor/stream/window/WindowProcessor.java), 
-it will create an appropriate expired event corresponding to the incoming current event with the expiring timestamp and store that 
-event in the window. The WindowProcessor will also forward the current event to the next Processor for further processing. 
-WindowProcessor using a scheduler or otherwise decides when to emit the events it has in memory and during that time it emits 
-those events as expired events, at times if its emitting all the events in the memory at ones it emits a single reset event interred of 
-sending one expired event for each event it has stored to reset the states in one go. From a window for each current event it has emitted it is essential 
-to emmit the same event as an expired event or a common reset event should be emitted.  
-This is vital in Siddhi because Siddhi relies on these events to calculate the Aggregations at the QuerySelector. 
-In QuerySelector, the arrived current events increase the aggregation values, expired events decrease the values, and 
+it creates an appropriate expired event corresponding to the incoming current event with the expiring timestamp, and stores that 
+event in the window. The `WindowProcessor`  also forwards the current event to the next processor for further processing. 
+t uses a scheduler or such to determine when to emit the events it has in memory. During that time, it emits these events as expired events. If it emits all the events in the memory at once, it emits a single reset event in one go instead of sending one expired event for each event it has stored to reset. For each current event emitted from a window, a corresponding expired event must be emitted. If not, a common reset event must be emitted.  
```

Comment:
It

Reply:
If it needs to emit all the events in the memory at once, it emits a single reset event interred of 
 -sending one expired event for each event it has stored such that it can reset the states in one go